src/pages/home/report/ReactionList/BaseReactionList.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable rulesdir/onyx-props-must-have-default */
+import Str from 'expensify-common/lib/str';
+import React from 'react';
+import {FlatList} from 'react-native';
+import type {FlatListProps} from 'react-native';
+import OptionRow from '@components/OptionRow';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import Navigation from '@libs/Navigation/Navigation';
+import * as UserUtils from '@libs/UserUtils';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type {PersonalDetails} from '@src/types/onyx';
+import HeaderReactionList from './HeaderReactionList';
+import type ReactionListProps from './types';
+
+type BaseReactionListProps = ReactionListProps & {
+    /**
+     *  Array of personal detail objects
+     */
+    users: PersonalDetails[];
+
+    /**
+     * Returns true if the current account has reacted to the report action (with the given skin tone).
+     */
+    hasUserReacted?: boolean;
+
+    /**
+     * Returns true if the reaction list is visible
+     */
+    isVisible?: boolean;
+};
+
+const keyExtractor: FlatListProps<PersonalDetails>['keyExtractor'] = (item, index) => `${item.login}+${index}`;
+
+const getItemLayout = (data: ArrayLike<PersonalDetails> | null | undefined, index: number): {length: number; offset: number; index: number} => ({
+    index,
+    length: variables.listItemHeightNormal,
+    offset: variables.listItemHeightNormal * index,
+});
+
+function BaseReactionList({hasUserReacted = false, users, isVisible = false, emojiCodes, emojiCount, emojiName, onClose}: BaseReactionListProps) {
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const {hoveredComponentBG, reactionListContainer, reactionListContainerFixedWidth, pv2} = useThemeStyles();
+
+    if (!isVisible) {
+        return null;
+    }
+
+    /**
+     * Given an emoji item object, render a component based on its type.
+     * Items with the code "SPACER" return nothing and are used to fill rows up to 8
+     * so that the sticky headers function properly
+     *
+     */
+    const renderItem: FlatListProps<PersonalDetails>['renderItem'] = ({item}) => (
+        <OptionRow
+            boldStyle
+            style={{maxWidth: variables.mobileResponsiveWidthBreakpoint}}
+            hoverStyle={hoveredComponentBG}
+            onSelectRow={() => {
+                onClose?.();
+
+                Navigation.navigate(ROUTES.PROFILE.getRoute(item.accountID));
+            }}
+            option={{
+                reportID: String(item.accountID),
+                text: Str.removeSMSDomain(item.displayName ?? ''),
+                alternateText: Str.removeSMSDomain(item.login ?? ''),
+                participantsList: [item],
+                icons: [
+                    {
+                        id: item.accountID,
+                        source: UserUtils.getAvatar(item.avatar, item.accountID),
```

Comment:
avatar and accountID is always expected to be present?

Reply:
I believe in its type (`PersonalDetails`) which comes from `src/types/onyx/index.ts`, it is mentioned that `avatar` can be `undefined`, unlike `accountID`. Even if the `avatar` is absent, the helper function (`UserUtils.getAvatar`) handles the situation to return the default avatar. Is it any further handling required in here? I believe all the emoji reactions must have an icon in the list.
Please let me know.

Reply:
Thanks for clarifying. Sounds good to me

Reply:
No problem. Thanks for your consideration. So I mark this conversation as resolved.